src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int mySize;
+    private T[] arrayList;
+
+    public ArrayList() {
+        mySize = 0;
+        arrayList = (T[]) new Object[DEFAULT_SIZE];
+    }
 
     @Override
     public void add(T value) {
-
+        checkSize(arrayList.length);
+        arrayList[mySize] = value;
+        mySize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkConditions(index, mySize + 1);
```

Comment:
let's move the mySize parameter from checkConditions and just make a different check for this method